plutus-core/testlib/PlutusCore/Generators/QuickCheck/Builtin.hs

```diff
@@ -184,14 +183,18 @@ genDataFromSpine els = oneof
             ]
     ]
 
-instance Arbitrary Data where
-    arbitrary = arbitrary >>= genDataFromSpine
+instance ArbitraryBuiltin Data where
+    arbitraryBuiltin = arbitrary >>= genDataFromSpine
+
+    shrinkBuiltin (Constr i ds) = ds ++ map (Constr i) (shrinkBuiltin ds)
+    shrinkBuiltin (Map ps)      = map fst ps ++ map snd ps ++ map Map (shrinkBuiltin ps)
+    shrinkBuiltin (List ds)     = ds ++ map List (shrinkBuiltin ds)
+    shrinkBuiltin (I i)         = map I (shrinkBuiltin i)
+    shrinkBuiltin (B b)         = I 0 : map B (shrinkBuiltin b)
 
-    shrink (Constr i ds) = ds ++ map (Constr i) (shrink ds)
```

Comment:
Values of built-in types should be shrunk with `shrinkBuiltin`, not `shrink`, so this was a minor bug in the shrinking machinery (that I introduced one PR ago).